### ISSUE TITLE
WeakRefStrings changes the behavior of `string`

### ISSUE THREAD

**Issue by bluesmoon** [JuliaData/InlineStrings.jl#2]
If I broadcast the `string` method to an empty `AbstractString[]`, it remains unchanged.

If I do the same after including `WeakRefStrings`, it gets changed to `Any[]`

Julia v1.6.3
WeakRefStrings v1.3.0

```julia
julia> string.(AbstractString[])
AbstractString[]

julia> using WeakRefStrings

julia> string.(AbstractString[])
Any[]

julia>
```

**Comment by bkamins** [comment]
> I'm also wondering if this is really that important.

I think it is important. In some workflows one dispatches on `eltype` of a column being a subtype of `AbstractString`.

**Comment by bluesmoon** [comment]
Yes, it affects our code. We use multiple dispatch to call the appropriate method based on element type.

ie, we have something like this:
```julia
function foo(col::AbstractVector{<:AbstractString})
   ...
end

function foo(col::AbstractVector{<:AbstractFloat})
   ...
end

function foo(col::AbstractVector{<:Integer})
   ...
end
```

If the array changes to `Any[]` then julia cannot find an appropriate method to call.

**Comment by quinnj** [comment]
Ok, makes sense. Fixes are up [here](https://github.com/JuliaData/WeakRefStrings.jl/pull/93) and [here](https://github.com/JuliaData/InlineStrings.jl/pull/7)
